File: src/main/java/com/jozufozu/exnihiloomnia/common/blocks/barrel/BlockBarrelStainedGlass.java
package com.jozufozu.exnihiloomnia.common.blocks.barrel;

import com.jozufozu.exnihiloomnia.common.lib.LibBlocks;
import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;
import net.minecraft.block.properties.PropertyEnum;
import net.minecraft.item.EnumDyeColor;
import net.minecraft.util.BlockRenderLayer;

public class BlockBarrelStainedGlass extends BlockBarrelColored
{
    public static final PropertyEnum<EnumDyeColor> COLOR = PropertyEnum.create("color", EnumDyeColor.class);
    
    public BlockBarrelStainedGlass()
    {
        super(LibBlocks.STAINED_GLASS_BARREL, Material.GLASS, SoundType.GLASS);
        this.setDefaultState(this.blockState.getBaseState().withProperty(COLOR, EnumDyeColor.WHITE));
        this.setHardness(0.7f);
    }
    
    @Override
    public BlockRenderLayer getBlockLayer()
    {
        return BlockRenderLayer.TRANSLUCENT;
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/blocks/barrel/BlockBarrelStainedGlass.kt
package com.jozufozu.exnihiloomnia.common.blocks.barrel

import com.jozufozu.exnihiloomnia.common.lib.LibBlocks
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material
import net.minecraft.item.EnumDyeColor
import net.minecraft.util.BlockRenderLayer

class BlockBarrelStainedGlass : BlockBarrelColored(LibBlocks.STAINED_GLASS_BARREL, Material.GLASS, SoundType.GLASS) {
    companion object {
        @JvmField val COLOR = BlockBarrelColored.COLOR
    }

    init {
        this.defaultState = this.blockState.baseState.withProperty(COLOR, EnumDyeColor.WHITE)
        this.setHardness(0.7f)
    }

    override fun getBlockLayer() = BlockRenderLayer.TRANSLUCENT
}
